packs/BP/scripts/network_links/network_link_component.ts

```diff
@@ -0,0 +1,57 @@
+import { BlockCustomComponent, world } from "@minecraft/server";
+import { getBlockIoCategories } from "../io";
+import { MachineNetwork } from "../network";
+import { registerScriptEventHandler } from "mcbe-addon-ipc";
+import { deserializeDimensionLocation, makeError, SerializableDimensionLocation } from "@/public_api/src/internal";
+import { NetworkLinkGetRequest, NetworkLinkGetResponse, NetworkLinkAddRequest, NetworkLinkRemoveRequest, NetworkLinkDestroyRequest } from "@/public_api/src/network_links/ipc_events";
+import { NetworkLinkNode } from "./network_link_internal";
+
+export const networkLinkComponent: BlockCustomComponent = {
+    onPlace(ev) {
+        const ioCategories = getBlockIoCategories(ev.block);
+        MachineNetwork.updateAdjacent(
+            ev.block,
+            ioCategories === "any" ? undefined : ioCategories
+        );
+    },
+
+    onPlayerDestroy(ev) {
+        const linkNode = NetworkLinkNode.tryGetAt(ev.dimension, ev.block.location);
+
+        // remove all incoming and outbound links to this node in the network
+        if (linkNode) linkNode.destroyNode();
+ 
+        // update the rest of the blocks in the network.
+        MachineNetwork.updateWithBlock(ev.block);
+    },
+};
+
+function _getNetwork(self: SerializableDimensionLocation) {
+    const location = deserializeDimensionLocation(self);
+    const block = location.dimension.getBlock(location);
+    if (!block) makeError(`_getNetwork failed to get block`);
+    return NetworkLinkNode.fromBlock(block);
+}
+
+registerScriptEventHandler<NetworkLinkGetRequest, NetworkLinkGetResponse>("fluffyalien_energisticscore:ipc.network_link_get", (payload => {
+    const link = _getNetwork(payload.self);
+    return { locations: link.getConnections() }; 
+}))
+
+registerScriptEventHandler<NetworkLinkAddRequest, void>("fluffyalien_energisticscore:ipc.network_link_add", (payload => {
```

Comment:
A handler is a two-way event, it should not return void. Use [registerScriptEventListener](https://fluffyalien1422.github.io/mcbe-addon-ipc/functions/registerScriptEventListener.html) to register a one-way listener

Reply:
This is required since `dispatchScriptEvent` is not asynchronous, and we need to be able to wait for the operation to be complete due to the delays of script events. Otherwise we could run into order of operation issues where the programmer is expecting the link to have been made successfully at this point, but in actuality it wont be done for a couple ticks.